### PR TITLE
Automatic page visiting

### PR DESCRIPTION
**Pull request by Lucki**
This isn't fully tested yet but should visit every url in the original post.

Also the missing commit from my last PR is included.

### CODE REVIEW

File: alienwarearena_helper.user.js [hunk 1]
```diff
@@ -905,6 +928,37 @@ Sorting from fresh ones to old ones.">Vote for newly uploaded ${sectionType}${(s
 		}
 	}
 
+	async function visitURL() {
+		try {
+			let dailyThreadId = await getDailyThread();
+			let result = await fetch('/ucf/show/' + dailyThread, {credentials: 'same-origin'});
+			let text = await response.text();
+			let postText = text.match(/<div class="discussion__op-content ucf__content">[\s\S]*<\/div>/);
```
Comment (thomas-ashcraft) [inline alienwarearena_helper.user.js:936]: Nit: those things can be done through DOMParser() (used it in postReplies() to check for post). Maybe you will find that way convenient. For example, you can do things like this:
``` JS
const parser = new DOMParser();
let doc = parser.parseFromString(text, 'text/html');
let someElement = doc.querySelector(`div.some-css-selector`);
let someLinkInsideElement = someElement.querySelector(`a`);
console.log('link is ', someLinkInsideElement.href)
```
The main advantage over RegExp here is that you are not dependent on the HTML code - DOMParser and querySelector will do things for you. And it a bit easier to maintain.

Comment (Lucki) [reply alienwarearena_helper.user.js:936]: I see. That's probably the experience you get when you're doing these more often.

Comment (Lucki) [reply alienwarearena_helper.user.js:936]: @thomas-ashcraft With these changes it's now working for me:

~~~ diff
diff --git a/alienwarearena_helper.user.js b/alienwarearena_helper.user.js
index 7eb8933..e1be66d 100644
--- a/alienwarearena_helper.user.js
+++ b/alienwarearena_helper.user.js
@@ -936,29 +936,42 @@ Sorting from fresh ones to old ones.">Vote for newly uploaded ${sectionType}${(s
 
 	async function visitURL() {
 		try {
+			const parser = new DOMParser();
 			let dailyThreadId = await getDailyThread();
-			let result = await fetch('/ucf/show/' + dailyThread, {credentials: 'same-origin'});
-			let text = await response.text();
-			let postText = text.match(/<div class="discussion__op-content ucf__content">[\s\S]*<\/div>/);
-			let urls = postText.match(/<a href="http.*?alienwarearena\.com(.*?)".*?<\/a>/g);
+			let result = await fetch(`/ucf/show/${dailyThreadId}`, {credentials: 'same-origin'});
+			let text = await result.text();
+			let doc = parser.parseFromString(text, 'text/html');
+			let opText = doc.querySelector(`div.discussion__op-content.ucf__content`);
+			let urls = opText.querySelectorAll(`a`);
 			for (let url of urls) {
 				try {
-					await fetch(url, {credentials: 'same-origin'});
-					let questDone = await dailyQuestDone();
-					if (questDone) {
-						ui.newStatusMessage('Successfully visited page!');
-						return;
+					let link = new URL(url.href);
+					let ids = link.pathname.match(/([0-9]+)/);
+					for (let id of ids) {
+						await fetch(link.pathname, {credentials: 'same-origin'});
+						await fetch(`/ucf/increment-views/${id}`, {
+							credentials: 'same-origin',
+							method: 'POST',
+							headers: {
+								'X-Requested-With': 'XMLHttpRequest'
+							}
+						});
+						let questDone = await dailyQuestDone();
+						if (questDone) {
+							ui.newStatusMessage('Successfully visited page!');
+							return;
+						}
 					}
 				} catch (err) {
 					continue;
 				}
 			}
 
-			document.location.href = '/ucf/show/' + dailyThreadId;
+			document.location.href = `/ucf/show/${dailyThreadId}`;
 		} catch (err) {
 			try {
 				let dailyThreadId = await getDailyThread();
-				document.location.href = '/ucf/show/' + dailyThreadId;
+				document.location.href = `/ucf/show/${dailyThreadId}`;
 			} catch (err) {
 				document.location.href = '/forums/board/113/awa-on-topic';
 			}
@@ -1042,7 +1055,7 @@ Sorting from fresh ones to old ones.">Vote for newly uploaded ${sectionType}${(s
 						<span class="more-link right"></span></a>`).appendTo(".quest-item > .col-2");
 					break;
 				case 'visit_page':
-					$(`<a class="btn btn-default awah-btn-quest" href="javascript:void(0);" data-awah-tooltip="Visit forum" data-awah-quest="visit">
+					$(`<a class="btn btn-default awah-btn-quest" href="javascript:void(0);" data-awah-tooltip="Automatic link visiting" data-awah-quest="visit">
 						<span class="more-link right"></span></a>`).appendTo(".quest-item > .col-2");
 					break;
 				case 'change_avatar_placeholder':
~~~

Comment (thomas-ashcraft) [reply alienwarearena_helper.user.js:936]: @Lucki  Feel free to make a pull request. I will gladly merge it. You're doing great!

**Pull request merged**